testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/VariablesCompletionTestCase.java

```diff
@@ -0,0 +1,146 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.test.integration.management.cli;
+
+import org.jboss.as.cli.CommandContext;
+import org.jboss.as.cli.CommandContextFactory;
+import org.jboss.as.cli.CommandLineException;
+import org.jboss.as.cli.impl.CommandContextConfiguration;
+import org.jboss.as.test.integration.management.util.CLITestUtil;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.io.ByteArrayOutputStream;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * WFCORE-1714 Test for Variables Completion
+ *
+ * @author wangc
+ */
+@RunWith(WildflyTestRunner.class)
+public class VariablesCompletionTestCase {
+
+    private static final String FOO_NAME = "foo";
+    private static final String FOO_VALUE = "/subsystem=logging";
+    private static final String FOOBAR_NAME = "foobar";
+    private static final String FOOBAR_VALUE = "/subsystem=logging/console-handler=CONSOLE";
+
+    private static ByteArrayOutputStream cliOut;
+    private static CommandContextConfiguration config;
+    private static CommandContext ctx;
+
+    @Before
+    public void setup() throws CommandLineException {
+        cliOut = new ByteArrayOutputStream();
+
+        System.setProperty("aesh.ansi","false");
+        System.setProperty("aesh.terminal","org.jboss.aesh.terminal.TestTerminal");
+        CLITestUtil.setJBossCliConfig();
+        config = new CommandContextConfiguration.Builder()
+                .setController(CLITestUtil.constructUri(null, TestSuiteEnvironment.getServerAddress(), TestSuiteEnvironment.getServerPort()))
+                .setUsername(null)
+                .setPassword(null)
+                .setConsoleInput(null)
+                .setConsoleOutput(cliOut)
+                .setDisableLocalAuth(false)
+                .setInitConsole(true)
+                .build();
+        ctx = CommandContextFactory.getInstance().newCommandContext(config);
+
+        // set both variables
+        ctx.handle("set " + FOO_NAME + "=" + FOO_VALUE);
+        ctx.handle("set " + FOOBAR_NAME + "=" + FOOBAR_VALUE);
+        cliOut.reset();
+
+        ctx.handle("echo $" + FOO_NAME);
+        assertTrue(cliOut.toString().contains(FOO_VALUE));
+        ctx.handle("echo $" + FOOBAR_NAME);
+        assertTrue(cliOut.toString().contains(FOOBAR_VALUE));
+
+        ctx.handle("connect"); // This is necessary to get : and /
```

Comment:
You should use the class CLIWrapper and call the constructor CLIWrapper(true). It will do all setup for you ;-)